cypress/integration/common/post.js

```diff
@@ -83,3 +86,74 @@ And("the post with title {string} has a ribbon for pinned posts", (title) => {
 Then("I see a toaster with {string}", (title) => {
   cy.get(".iziToast-message").should("contain", title);
 })
+
+Then("I should be able to {string} a teaser image", condition => {
+  cy.reload()
+  let teaserImageUpload = "onourjourney.png";
+  if (condition === 'change') teaserImageUpload = "humanconnection.png";
+  cy.fixture(teaserImageUpload).as('postTeaserImage').then(function() {
+    cy.get("#postdropzone").upload(
+      { fileContent: this.postTeaserImage, fileName: teaserImageUpload, mimeType: "image/png" },
+      { subjectType: "drag-n-drop", force: true }
+    );
+  })
+})
+
+Then('confirm crop', () => {
+  cy.get('.crop-confirm')
+    .click()
+})
+
+Then("I add all required fields", () => {
+  cy.get('input[name="title"]')
+    .type('new post')
+    .get(".editor .ProseMirror")
+    .type('new post content')
+    .get(".categories-select .base-button")
+    .first()
+    .click()
+    .get('.ds-flex-item > .ds-form-item .ds-select ')
+    .click()
+    .get('.ds-select-option')
+    .eq(languages.findIndex(l => l.code === 'en'))
+    .click()
+})
+
+Then("the post was saved successfully with the {string} teaser image", condition => {
+  if (condition === 'updated') 
+    expectedValue = { title: 'to be updated', content: 'successfully updated', src: 'humanconnection' }
+  cy.get(".ds-card-content > .ds-heading")
+    .should("contain", expectedValue.title)
+    .get(".content")
+    .should("contain", expectedValue.content)
+    .get('.post-page img')
+    .should("have.attr", "src")
+    .and("contains", expectedValue.src)
+})
+
+Then("the first image should be removed from the preview", () => {
+  cy.fixture("humanconnection.png").as('postTeaserImage').then(function() {
+    cy.get("#postdropzone")
+      .children()
+      .get('img.thumbnail-preview')
+      .should('have.length', 1)
+      .and('have.attr', 'src')
+      .and('contain', this.postTeaserImage)
+  })
+})
+
+Then('the post was saved successfully without a teaser image', () => {
+  cy.get(".ds-card-content > .ds-heading")
+    .should("contain", 'new post')
+    .get(".content")
+    .should("contain", 'new post content')
+    .get('.post-page')
+    .should('exist')
+    .get('.post-page img.ds-card-image')
+    .should('not.exist')
+})
+
+Then('I should be able to remove it', () => {
+  cy.get('.crop-cancel')
+    .click()
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-14T11:45:59Z" title="Friday, February 14th 2020, 12:45:59 pm +01:00">Feb 14, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -83,3 +86,71 @@ And("the post with title {string} has a ribbon for pinned posts", (title) => {
 Then("I see a toaster with {string}", (title) => {
   cy.get(".iziToast-message").should("contain", title);
 })
+
+Then("I should be able to {string} a teaser image", condition => {
+  let teaserImageUpload = "onourjourney.png";
+  if (condition === 'change') teaserImageUpload = "humanconnection.png";
+  cy.fixture(teaserImageUpload).as('postTeaserImage').then(function() {
+    cy.get("#postdropzone").upload(
+      { fileContent: this.postTeaserImage, fileName: teaserImageUpload, mimeType: "image/png" },
+      { subjectType: "drag-n-drop", force: true }
+    );
+  })
+})
+
+Then('confirm crop', () => {
+  cy.get('.crop-confirm')
+    .click()
+})
+
+Then("I add all required fields", () => {
+  cy.get('input[name="title"]')
+    .type('new post')
+    .get(".editor .ProseMirror")
+    .type('new post content')
+    .get(".base-button")
+    .contains("Just for Fun")
+    .click()
+    .get('.ds-flex-item > .ds-form-item .ds-select ')
+    .click()
+    .get('.ds-select-option')
+    .eq(languages.findIndex(l => l.code === 'en'))
+    .click()
+})
+
+Then("the post was saved successfully with the {string} teaser image", condition => {
+  cy.get(".ds-card-content > .ds-heading")
+    .should("contain", condition === 'new' ? 'new post' : 'to be updated')
+    .get(".content")
+    .should("contain", condition === 'new' ? 'new post content' : 'successfully updated')
+    .get('.post-page img')
+    .should("have.attr", "src")
+    .and("contains", condition === 'new' ? "onourjourney" : "humanconnection")
+})
+
+Then("the first image should be removed from the preview", () => {
+  cy.fixture("humanconnection.png").as('postTeaserImage').then(function() {
+    cy.get("#postdropzone")
+      .children()
+      .get('img.thumbnail-preview')
+      .should('have.length', 1)
+      .and('have.attr', 'src')
+      .and('contain', this.postTeaserImage)
+  })
+})
+
+Then('the post was saved successfully without a teaser image', () => {
+  cy.get(".ds-card-content > .ds-heading")
+    .should("contain", 'new post')
+    .get(".content")
+    .should("contain", 'new post content')
+    .get('.post-page')
+    .should('exist')
+    .get('.post-page img.ds-card-image')
+    .should('not.exist')
+})
+
+Then('I should be able to remove it', () => {
+  cy.get('.crop-cancel')
+    .click()
```


Already clicking on it? You don't want to check the effect?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-14T11:57:15Z" title="Friday, February 14th 2020, 12:57:15 pm +01:00">Feb 14, 2020</time>_

---


what do you mean? like check that there is a `thumbnail-preview`?